lib/CXGN/BrAPI/v2/Lists.pm

```diff
@@ -35,33 +36,51 @@ sub search {
 	my @data;
 	my $lists;
 
-	if ($user_id){
-		$lists = CXGN::List::available_lists($self->bcs_schema()->storage->dbh(),$user_id,$types_arrayref->[0]);
-	} else {
-		$lists = CXGN::List::available_public_lists($self->bcs_schema()->storage->dbh(),$types_arrayref->[0]);
+	$lists = CXGN::List::all_lists($self->bcs_schema()->storage->dbh(),$user_id,$types_arrayref->[0]);
```

Comment:
the `convert_to_breedbase_type` method needs to be called to convert the $types_arrayref->[0] value.

Also, a check should be done to either validate the listType is one of the ones defined in BrAPI OR check if there are any results from this call before proceeding.  When testing, I sent in an invalid type, and got a 500 error because the downstream SQL query used an `in` comparator, but the contents of the `in` were empty: ex: `select * from list where list_id in ()` <- invalid because there are no values in the parenthesis.

Reply:
converted to breedbase types and added a list type check in that returns a 400 if the type isn't found.